File: tasks/builtin/typescript/typescript_refactor_001/workspace/tests/test_baseline.js
const assert = require('assert');
const { Dispatcher } = require('../src/dispatcher');

async function run() {
  const d = new Dispatcher();
  const res = await d.dispatch(async () => 'success');
  assert.strictEqual(res, 'success');
  console.log('Baseline passed');
}

run();

File: tasks/builtin/python/python_async_001/workspace/tests/test_baseline.py
import asyncio
from src.pipeline import execute_batch


def test_baseline_all_successful():
    async def runner():
        async def worker(val):
            await asyncio.sleep(0.001)
            return val * 2

        coros = [worker(1), worker(2), worker(3)]
        res = await execute_batch(coros)
        assert res["successes"] == [2, 4, 6]
        assert len(res["failures"]) == 0

    asyncio.run(runner())
